packages/@react-aria/test-utils/src/user.ts

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2024 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {pointerMap} from './';
+import {SelectTester} from './select';
+import {TableTester} from './table';
+import userEvent from '@testing-library/user-event';
+
+interface UserOpts {
+  interactionType?: 'mouse' | 'touch' | 'keyboard'
+}
+
+export class User {
+  select: SelectTester;
+  table: TableTester;
+
+  constructor(opts: UserOpts = {}) {
+    let {interactionType} = opts;
+    let user = userEvent.setup({delay: null, pointerMap});
+    this.select = new SelectTester({user, interactionType});
+    this.table = new TableTester({user, interactionType});
```

Comment:
maybe make factory function in this object that returns a new instance of a specified role instead of initializing them here to avoid breaks due to shared internal state if a test breaks